server/client_server.go

```diff
@@ -22,10 +30,116 @@ func newServerClient(jiraClient *jira.Client) Client {
 	}
 }
 
+type IssueInfo struct {
+	Values []*jira.MetaIssueType `json:"values,omitempty"`
+}
+
+type FieldInfo struct {
+	Values []interface{} `json:"values,omitempty"`
+}
+
+type FieldValues struct {
+	FieldID string `json:"fieldId,omitempty"`
+}
+
+type FieldID struct {
+	Values []FieldValues `json:"values,omitempty"`
+}
+
+type Version struct {
+	VersionInfo string `json:"version,omitempty"`
+}
+
 // GetCreateMeta returns the metadata needed to implement the UI and validation of
 // creating new Jira issues.
 func (client jiraServerClient) GetCreateMeta(options *jira.GetQueryOptions) (*jira.CreateMetaInfo, error) {
-	cimd, resp, err := client.Jira.Issue.GetCreateMetaWithOptions(options)
+	v := new(Version)
+	req, err := client.Jira.NewRequest(http.MethodGet, ServerInfoApiEndpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err = client.Jira.Do(req, v); err != nil {
+		return nil, err
+	}
+
+	currentVersion, err := version.NewVersion(v.VersionInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	pivotVersion, err := version.NewVersion(PivotVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	var info *jira.CreateMetaInfo
+	var resp *jira.Response
+	if currentVersion.LessThan(pivotVersion) {
+		info, resp, err = client.Jira.Issue.GetCreateMetaWithOptions(options)
+	} else {
+		cd, response, apiErr := client.Jira.Project.ListWithOptions(options)
+		meta := new(jira.CreateMetaInfo)
+
+		if apiErr == nil {
+			for i := 0; i < len(*cd); i++ {
+				meta.Expand = (*cd)[i].Expand
+				apiEndpoint := fmt.Sprintf("%s%s/issuetypes", CreateMetaAPIEndpoint, (*cd)[i].ID)
+				req, err = client.Jira.NewRequest(http.MethodGet, apiEndpoint, nil)
+				if err != nil {
+					break
+				}
+
+				issues := new(IssueInfo)
+				response, err = client.Jira.Do(req, issues)
+				if err != nil {
+					break
+				}
+
+				project := &jira.MetaProject{
+					Expand:     (*cd)[i].Expand,
+					Self:       (*cd)[i].Self,
+					Id:         (*cd)[i].ID,
+					Key:        (*cd)[i].Key,
+					Name:       (*cd)[i].Name,
+					IssueTypes: issues.Values,
+				}
+
+				for _, issue := range project.IssueTypes {
+					apiEndpoint := fmt.Sprintf("%s%s/issuetypes/%s", CreateMetaAPIEndpoint, (*cd)[i].ID, issue.Id)
+					req, err = client.Jira.NewRequest(http.MethodGet, apiEndpoint, nil)
```

Comment:
Instead of first creating the request and then calling the `Do` function, you can also directly use the `Get` function.
For eg - 
```
client.Jira.Get(apiEndpoint)
```

Reply:
client.Jira.Get(apiEndpoint) is undefined as `it has no field or method Get`